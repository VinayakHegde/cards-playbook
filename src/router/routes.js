export default [
  {
    path: "/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/new",
    name: "Home",
    meta: {
      header: "CARDS"
    },
    component: () =>
      import(/* webpackChunkName: "home" */ "../views/home/Home.vue")
  },
  {
    path: "/deck/:hash",
    name: "Pack",
    meta: {
      header: "Ordered Pile"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/pack/Pack.vue"),
    props: true
  },
  {
    path: "/error",
    name: "Error",
    meta: {
      header: "404 | Page not found"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/error/Error.vue"),
    props: true
  },
  {
    path: "*",
    redirect: {
      name: "Error"
    }
  }
];
