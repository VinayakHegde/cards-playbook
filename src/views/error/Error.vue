<script>
import { mapActions } from "vuex";
export default {
  created() {
    this.setloader(false);
  },
  methods: {
    ...mapActions({
      setloader: "setLoader"
    })
  }
};
</script>
<template>
  <main>
    <router-link to="/">Back Home</router-link>
  </main>
</template>
<style lang="scss" scoped>
main {
  text-align: center;
}
</style>
