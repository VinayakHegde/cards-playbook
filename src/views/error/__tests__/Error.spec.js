import Vuex from "vuex";
import { mount, createLocalVue } from "@vue/test-utils";
import VueRouter from "vue-router";
import routes from "@/router/routes";
import Error from "@/views/error/Error.vue";
import storeConfig from "@/store/store-config";

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
describe("Route.vue", () => {
  it("match the snapshot", () => {
    const router = new VueRouter({ routes });
    const store = new Vuex.Store(storeConfig);
    const wrapper = mount(Error, {
      localVue,
      router,
      store
    });
    expect(wrapper).toMatchSnapshot();
    wrapper.find("a").trigger("click");

    localVue.nextTick(() => {
      expect(wrapper.vm.$route.fullPath).toBe("/");
    });
  });
});
