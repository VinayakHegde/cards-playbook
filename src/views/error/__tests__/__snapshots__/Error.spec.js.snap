// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route.vue match the snapshot 1`] = `<main><a href="#/" aria-current="page" class="router-link-exact-active router-link-active">Back Home</a></main>`;
